src/components/CustomMessage.tsx

```diff
@@ -18,53 +17,32 @@ import UserMessageWithBodyInput from './UserMessageWithBodyInput';
 import { LOCAL_MESSAGE_CUSTOM_TYPE } from '../const';
 import { useConstantState } from '../context/ConstantContext';
 import { useWidgetSession } from '../context/WidgetSettingContext';
+import { TypingBubble } from '../foundation/components/TypingBubble';
 import { getSourceFromMetadata, parseTextMessage, Token } from '../utils';
 import { messageExtension } from '../utils/messageExtension';
-import { isLastMessageInStreaming, isLocalMessageCustomType, isSentBy } from '../utils/messages';
+import { isLocalMessageCustomType, isSentBy } from '../utils/messages';
 
 type Props = {
   message: CoreMessageType;
   activeSpinnerId: number;
   botUser?: User;
-  isBotWelcomeMessage: boolean;
-  isLastBotMessage: boolean;
-  messageCount: number;
   chainTop?: boolean;
   chainBottom?: boolean;
 };
 
 export default function CustomMessage(props: Props) {
-  const {
-    message,
-    activeSpinnerId,
-    botUser,
-    chainTop,
-    chainBottom,
-    isBotWelcomeMessage,
-    isLastBotMessage,
-    messageCount,
-  } = props;
-  const commonProps = {
-    chainTop,
-    chainBottom,
-    isBotWelcomeMessage,
-    isLastBotMessage,
-    messageCount,
-    message,
-  };
+  const { message, activeSpinnerId, botUser } = props;
   const { replacementTextList, enableEmojiFeedback, botStudioEditProps } = useConstantState();
   const { userId: currentUserId } = useWidgetSession();
   const { profileUrl } = botStudioEditProps?.botInfo ?? {};
+
   const botUserId = botUser?.userId;
   const botProfileUrl = profileUrl ?? botUser?.profileUrl ?? '';
   const isWaitingForBotReply = activeSpinnerId === message.messageId && !!botUser;
 
   const shouldRenderFeedback = () => {
     return (
-      enableEmojiFeedback &&
-      message.myFeedbackStatus !== 'NOT_APPLICABLE' &&
-      !isBotWelcomeMessage &&
-      !(isLastBotMessage && isLastMessageInStreaming(message.data))
+      enableEmojiFeedback && message.myFeedbackStatus !== 'NOT_APPLICABLE' && !messageExtension.isStreaming(message)
```

Comment:
Isn't this too long for a single line? I prefer 

```
enableEmojiFeedback
&& message.myFeedbackStatus !== 'NOT_APPLICABLE'
&& !messageExtension.isStreaming(message)
```
What do you think of this rule?

Reply:
It follows the settings in prettier and eslint.